assets/src/edit-story/components/devTools/devTools.js

```diff
@@ -0,0 +1,210 @@
+            newElement.resource.poster =
+              'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAIAAAACCAYAAABytg0kAAAAGElEQVQYV2NkYGD4////fwZGEMnIyMgAAEvqB/6wfXLSAAAAAElFTkSuQmCC';
+
+            const videoEl = document.getElementById(`video-${newElement.id}`);
+            if (videoEl) {
+              videoEl.load();
+            }
+          }
+          if (element.type === 'image') {
+            newElement.resource.mimeType = 'image/png';
+            newElement.resource.src =
+              'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAIAAAACCAYAAABytg0kAAAAGElEQVQYV2NkYGD4////fwZGEMnIyMgAAEvqB/6wfXLSAAAAAElFTkSuQmCC';
+          }
+        }
+        return newElement;
+      }),
+    })),
+  };
+};
+
+function DevTools() {
+  const [isVisible, setIsVisible] = useState(false);
+  const [isBase64, setIsBase64] = useState(false);
+  const [isDummyResources, setIsDummyResources] = useState(false);
+  const { showSnackbar } = useSnackbar();
+  const textareaRef = useRef();
+  const {
+    state: { reducerState },
+    actions: { restore },
+  } = useStory();
+  // This will not work here
+  // const {
+  //   state, // stale data
+  //   internal: { restore }, // will not update
+  // } = useStoryReducer();
+
+  const {
+    pages,
+    current,
+    selection,
+    story: { stylePresets },
+  } = reducerState;
+  const reducerStateSlice = {
+    current,
+    selection,
+    story: { stylePresets },
+    pages,
+  };
+  const storyData = isDummyResources
+    ? replaceResourcesWithDummy(reducerStateSlice)
+    : reducerStateSlice;
+
+  const toggleDummyResources = () => setIsDummyResources((v) => !v);
+  const toggleBase64 = () => setIsBase64((v) => !v);
+  const toggleVisible = () => setIsVisible((v) => !v);
+  const copyToClipboard = () => {
+    textareaRef.current.select();
+    document.execCommand('copy');
+    textareaRef.current.setSelectionRange(0, 0);
+    showSnackbar({ message: 'Copied to clipboard', timeout: 1200 });
+  };
+  const loadFromInput = () => {
+    let input = output;
+    try {
+      input = window.atob(input);
+      // eslint-disable-next-line no-empty
+    } catch (e) {}
+    const inputState = JSON.parse(input);
+    const stateToRestore = {
+      ...inputState,
+      story: { ...reducerState.story, ...inputState.story },
+      capabilities: reducerState.capabilities,
```

Comment:
Is the reducer smart to keep all fields like `capabilities` if the props are missing in the `restore()` call? I think that'd be preferred.

Reply:
No, should we change it app-wide?

```js
function restore(state, { pages, current, selection, story, capabilities }) {
  if (!Array.isArray(pages) || pages.length === 0) {
    return state;
  }

  const newStory = typeof story === 'object' ? story : {};
  const newCapabilities = typeof capabilities === 'object' ? capabilities : {};
  const newCurrent = pages.some(({ id }) => id === current)
    ? current
    : pages[0].id;
  const newSelection = Array.isArray(selection) ? selection : [];
```

Reply:
I took a look and I think for this function specifically it'd be a bad idea. It's meant to be a "set full state". Our issue is truly only because `capabilities` do not belong there. So I'd rather work it around, as you're doing now.